nio_udp.c

```diff
@@ -25,6 +25,7 @@
 #include <arpa/inet.h>
 #include <limits.h>
 #include <netdb.h>
+#include <netinet/in.h>
```

Comment:
This should compile on Linux. Not sure about Cygwin.

I would check the above first and if this doesn't work then add a condition like:

```
#ifdef __FreeBSD__
  #include <netinet/in.h>
#endif
```